### DISCUSSION TITLE
How to handle pasting of HTML content into PortableText editor?

### DISCUSSION THREAD

**Discussion by lbittner-pdftron** [sanity-io/sanity#4987] [Sanity Studio Support]
Hey there!

We are utilizing the PortableText input provided by Sanity, but are having some issues when it comes to handling pasted HTML content.

We are migrating an old static site to Sanity which involves a lot of copy pasting text from a website into the portable text editor. When an HTML link (<a>) is pasted into the editor, Sanity automatically creates a "link" mark with a "href" field. The issue is that we are using the internalLink / externalLink pattern documented [here](https://www.sanity.io/guides/portable-text-internal-and-external-links), so our "link" type doesn't have an 'href' field. This is causing the following error when an HTML link is pasted into the editor:

![image](https://github.com/sanity-io/sanity/assets/30271920/3fc3ecc0-d557-4061-a5de-4adca7afa64f)

What I would like to have happen is to take the 'href' field it assigns and automatically set that value as the "externalLink" value instead. Is it possible to transform data in this way?

**Comment by lbittner-pdftron** [comment]
I should mention that I have tried some stuff with the `onPaste` prop that exists on `<PortableTextInput/>`, however handling it this way seems far too complex. When just a single HTML element is pasted, its easy enough to handle, but if a big block of HTML is pasted I now have to deal with parsing and creating many blocks which I feel is something I don't want to dive into.

I would prefer to handle this on the type/field level rather than the editor level.

**Reply by Msigvartsen** [reply]
Did you find any answer to this? Struggling with the same issue at the emoment

**Reply by haroldao** [reply]
Any updates?
